dafni_cli/model/inputs.py

```diff
@@ -0,0 +1,193 @@
+        description (str): Description of the model parameter
+        default (Any): Default value of the parameter
+    """
+
+    name: str
+    type: str
+    title: str
+    required: bool
+    description: str
+    default: Optional[Any] = None
+
+    # TODO: Left over from refactor, may no longer be needed
+    # (printed in format_parameters in ModelInputs but untouched for 2 years)
+    min: Optional[str] = None
+    max: Optional[str] = None
+
+    _parser_params: ClassVar[List[ParserParam]] = [
+        ParserParam("name", "name", str),
+        ParserParam("type", "type", str),
+        ParserParam("title", "title"),
+        ParserParam("required", "required"),
+        ParserParam("description", "description"),
+        ParserParam("default", "default"),
+        ParserParam("min", "min"),
+        ParserParam("max", "max"),
+    ]
+
+
+@dataclass
+class ModelInputs(ParserBaseObject):
+    """Dataclass representing inputs for a DAFNI model
+
+    Attributes:
+        parameters (List[ModelParameters]): List of parameters for the model
+        dataslots (List[ModelDataslot]): List of dataslots for the model
+    """
+
+    parameters: List[ModelParameter]
+    dataslots: List[ModelDataslot] = field(default_factory=list)
+
+    _parser_params: ClassVar[List[ParserParam]] = [
+        ParserParam("parameters", "parameters", ModelParameter),
+        ParserParam("dataslots", "dataslots", ModelDataslot),
+    ]
+
+    # TODO: Couldn't we use tabulate or something for this?
+    @staticmethod
+    def params_table_header(title_column_width: int, default_column_width: int) -> str:
+        """Formats the header row and the dashed line for an input parameters
+           table
+
+        Args:
+            title_column_width (int): Width of the title column to fit the
+                                      longest title (or "title") plus a buffer
+            default_column_width (int): Width of the default column to fit the
+                                        longest default (or "default") plus a
+                                        buffer
+
+        Returns:
+            str: Formatted string for the header and dashed line of the
+                 parameters table
+        """
+
+        header = (
+            f"{INPUT_TITLE_HEADER:{title_column_width}}"
+            f"{INPUT_TYPE_HEADER:{INPUT_TYPE_COLUMN_WIDTH}}"
+            f"{INPUT_MIN_HEADER:{INPUT_MIN_MAX_COLUMN_WIDTH}}"
+            f"{INPUT_MAX_HEADER:{INPUT_MIN_MAX_COLUMN_WIDTH}}"
+            f"{INPUT_DEFAULT_HEADER:{default_column_width}}"
+            f"{INPUT_DESCRIPTION_HEADER}\n"
+            + "-"
+            * (
+                title_column_width
+                + INPUT_TYPE_COLUMN_WIDTH
+                + 2 * INPUT_MIN_MAX_COLUMN_WIDTH
+                + default_column_width
+                + INPUT_DESCRIPTION_LINE_WIDTH
+            )
+            + "\n"
+        )
+        return header
+
+    def format_parameters(self) -> str:
```

Comment:
Potentially a later piece of work but what happens if there are no input params?

Reply:
Yeah, I think this should be changed it currently looks like
![image](https://user-images.githubusercontent.com/90245114/234215195-79ec8ec1-9499-4824-ba2a-d56f46cdd6ab.png)

I think this should be a separate issue as I think there could be others as well.